contracts/settling_game/modules/food/library.cairo

```diff
@@ -0,0 +1,192 @@
+# -----------------------------------
+# ____Food Library
+#   Food
+#
+# MIT License
+# -----------------------------------
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.starknet.common.syscalls import get_block_timestamp
+from starkware.cairo.common.uint256 import Uint256, uint256_eq
+from starkware.cairo.common.bool import TRUE, FALSE
+from starkware.cairo.common.math import unsigned_div_rem, assert_not_zero, assert_le, assert_nn
+from starkware.cairo.common.math_cmp import is_le
+
+from contracts.settling_game.utils.constants import (
+    FARM_LENGTH,
+    MAX_HARVESTS,
+    HARVEST_LENGTH,
+    STORE_HOUSE_SIZE,
+)
+from contracts.settling_game.utils.game_structs import (
+    RealmData,
+    RealmBuildingsIds,
+    HarvestType,
+    FoodBuildings,
+)
+from contracts.settling_game.utils.constants import SHIFT_41
+from contracts.settling_game.utils.general import unpack_data
+
+namespace Food:
+    # @notice Calculates how many available farms to harvest
+    # MAX_HARVESTS = max amount you can harvest at once.
+    # if more farms than MAX, return MAX and decayed farms. How will loose these harvests.
+    func calculate_harvest{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        time_since_update : felt
+    ) -> (total_farms : felt, total_remaining : felt, decayed_farms : felt):
+        alloc_locals
+
+        # TODO add max days can accure
+        let (total_farms, remaining_crops) = unsigned_div_rem(time_since_update, HARVEST_LENGTH)
+
+        let (le_max_farms) = is_le(total_farms, MAX_HARVESTS + 1)
+
+        if le_max_farms == TRUE:
+            return (total_farms, remaining_crops, 0)
+        end
+
+        return (MAX_HARVESTS, remaining_crops, total_farms - MAX_HARVESTS)
+    end
+
+    # @notice Calculates base food in the storehouse
+    func calculate_food_in_store_house{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    }(current_food_supply : felt, block_timestamp : felt) -> (current : felt):
+        alloc_locals
+
+        let current = current_food_supply - block_timestamp
```

Comment:
Same thing as before in `calculate_harvest` - this can be calculated in the function call and the func can accept only a single arg.